src/test/kotlin/graphics/scenery/tests/examples/basic/DFTExample.kt

```diff
@@ -0,0 +1,101 @@
+package graphics.scenery.tests.examples.basic
+
+import org.joml.Vector3f
+import graphics.scenery.*
+import graphics.scenery.backends.Renderer
+import graphics.scenery.numerics.Random
+import graphics.scenery.textures.Texture
+import graphics.scenery.utils.Image
+import graphics.scenery.volumes.Colormap
+import graphics.scenery.volumes.TransferFunction
+import graphics.scenery.volumes.Volume
+import net.imglib2.type.numeric.integer.UnsignedByteType
+import java.nio.file.Paths
+import kotlin.concurrent.thread
+import kotlin.math.pow
+
+/**
+ * <Description>
+ *  Visualizes a DFT snapshot.
+ *
+ * @author Lenz Fiedler <l.fiedler@hzdr.de>
+ */
+class DFTExample : SceneryBase("DFTExample", wantREPL = System.getProperty("scenery.master", "false").toBoolean()) {
+    override fun init() {
+        renderer = hub.add(SceneryElement.Renderer,
+            Renderer.createRenderer(hub, applicationName, scene, 512, 512))
+        val snapshot = DFTParser()
+
+        snapshot.parseCube(getDemoFilesPath() + "/volumes/dft/Fe_snapshot0_dens.cube")
+
+        // Scales the DFT coordinates (which are in Bohr units) for a better VR experience.
+        val scalingFactor = 0.5f
+
+        // Visualize the atoms.
+        val atomicRadius = 0.5f*scalingFactor
+        for(i in 0 until snapshot.numberOfAtoms) {
+            val s = Icosphere(atomicRadius*scalingFactor, 4)
+            // Shift the positions since the positions from the cube file are centers.
+            s.position = (snapshot.atomicPositions[i].mul(scalingFactor))
+            s.material.metallic = 0.3f
+            s.material.roughness = 0.9f
+            scene.addChild(s)
+        }
+
+        // Visualize the density data.
+
+        val volume = Volume.fromBuffer(emptyList(), snapshot.gridDimensions[0], snapshot.gridDimensions[1],
+                                        snapshot.gridDimensions[2], UnsignedByteType(), hub)
```

Comment:
Now I feel smart because I predicted your electron density is not actually UInt but UByte 🤣

Reply:
:smile: Yes, you're absolutely right there!